packages/core/src/mcp/oauth-provider.ts

```diff
@@ -89,10 +90,38 @@ interface PKCEParams {
   state: string;
 }
 
-const REDIRECT_PORT = 7777;
 const REDIRECT_PATH = '/oauth/callback';
 const HTTP_OK = 200;
 
+/**
+ * Get an available port for the OAuth callback server.
+ * Uses the same approach as the Google OAuth implementation.
+ *
+ * @returns A promise that resolves to an available port number
+ */
+function getAvailablePort(): Promise<number> {
+  return new Promise((resolve, reject) => {
+    const portStr = process.env['OAUTH_CALLBACK_PORT'];
+    if (portStr) {
+      const port = parseInt(portStr, 10);
+      if (isNaN(port) || port <= 0 || port > 65535) {
+        return reject(
+          new Error(`Invalid value for OAUTH_CALLBACK_PORT: "${portStr}"`),
+        );
+      }
+      return resolve(port);
+    }
+
+    const server = net.createServer();
```

Comment:
This ends up launching an http server twice. Can you fold this logic into the callback server launch? Have that launch on a randomized port and then extract the selected port from an event handler there.

Reply:
good catch, I will implement a fix that launches just a single http server

Reply:
Done